test/taskrun_test.go

```diff
@@ -210,3 +225,131 @@ spec:
 		t.Fatalf("-got, +want: %v", d)
 	}
 }
+
+// TestTaskRunModification is an exclusive test for SPIRE integration into taskrun.
+// The test starts a taskrun which has a sleep. While the taskrun is "sleep"ing,
+// the text modifies the taskrun results.
+// This change is caught by the taskrun reconciler when it tries to verify the results.
+func TestTaskRunModification(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var c *clients
+	var namespace string
+	var oldConfigMap map[string]string
```

Comment:
You might want to reset `results-from` to `sidecar-logs` in oldConfigMap since the default is `termination-message`

Reply:
setupWithFlags will save the oldFlags and resetFlags will reset the old flags after execution.
If we set resuls-from to sidecar-logs in reset, we will incorrectly be setting so even in beta mode.